baseplate/core.py

```diff
@@ -132,305 +136,177 @@ def from_upstream(cls, trace_id, parent_id, span_id, sampled, flags):
-
-        return self.payload.get("client_type", None)
-
-
-class UndefinedSecretsException(Exception):
-    """Exception raised when no SecretsStore is defined during token parsing
-
-    Occurs in the :py:class:`AuthenticationContext` object when it attempts to
-    parse an authentication payload without having a :py:class:`SecretsStore`
-    defined to provide the necessary secrets values to correctly decrypt and
-    parse the token.
-    """
-    def __init__(self):
-        super(UndefinedSecretsException, self).__init__(
-            "No SecretsStore defined for Authentication token parsing.")
-
-
-class WithheldAuthenticationError(Exception):
-    """Error raised when attempting to read from an unset authentication token
-
-    Occurs either because of a badly instantiated context or missing header
-    coming from upstream requests.
-    """
-    def __init__(self):
-        super(WithheldAuthenticationError, self).__init__(
-            "No Authentication token provided for this context.")
+        raise NoAuthenticationError
 
 
 _User = collections.namedtuple(
-    "_User", ["authentication_context", "loid", "cookie_created_ms"])
+    "_User", ["authentication_token", "loid", "cookie_created_ms"])
 _OAuthClient = collections.namedtuple(
-    "_OAuthClient", ["authentication_context"])
+    "_OAuthClient", ["authentication_token"])
 Session = collections.namedtuple("Session", ["id"])
 
 
 class User(_User):
-    """Wrapper for the user values in AuthenticationContext and the LoId cookie.
+    """Wrapper for the user values in AuthenticationToken and the LoId cookie.
     """
 
-    def event_fields(self):
-        """Dictionary of values to be added to events in the current context
-        """
-        if self.is_logged_in:
-            user_id = self.id
-        else:
-            user_id = self.loid
-        return {
-            "user_id": user_id,
-            "user_logged_in": self.is_logged_in,
-            "cookie_created": self.cookie_created_ms,
-        }
-
     @property
     def id(self):
-        """Authenticated account_id for the current User
+        """Authenticated account_id for the current User.
 
         :type: account_id string or None if context authentication is invalid
-        :raises: :py:class:`UndefinedSecretsException` if the
-            :py:class:`SecretsStore` has not been bound to the context handling
-            class (means that the authentication payload could not be decrypted
-            and the user_id returned)
-        :raises: :py:class:`WithheldAuthenticationError` if there was no
-            authentication token defined for the current context
+        :raises: :py:class:`NoAuthenticationError` if there was no
+            authentication token, it was invalid, or the subject is not an
+            account.
 
         """
-        return self.authentication_context.account_id
+        subject = self.authentication_token.subject
+        if not subject.startswith("t2_"):
```

Comment:
fixed the anon auth case. as for the logged in case and NoAuthenticationError, I think this is probably clearer since the user should always check is_logged_in first?